src/Native/System.Native/pal_uid.cpp

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "../config.h"
+#include "pal_uid.h"
+
+#include <assert.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <pwd.h>
+
+/**
+* Gets a password structure for the given uid.
+* Implemented as shim to getpwuid_r(3).
+*
+* Returns 0 for success, -1 for failure. Sets errno on failure.
+*/
+extern "C"
+int32_t GetPwUid(
+	int32_t         uid,
+	struct Passwd  *pwd,
```

Comment:
Nit 1: 'struct' isn't necessary here as we're using C++. 

Nit 2: I prefer Passwd* pwd over Passwd *pwd.

Nit disclaimer: These are just my personal preferences and we haven't really set C++ coding guidelines.